### PR TITLE
Use `config.layer_norm_eps` in some `nn.LayerNorm`.

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

⚠️⚠️⚠️ This changes the `eps` of those LayerNorm layers from (the default) `1e-5` to `1e-12`, and the outputs will have slightly differences before/after this PR. ⚠️⚠️⚠️

----

Similar to #20554, but this time instead of removing the attribute from config, we use `config.layer_norm_eps` in some `nn.LayerNorm`.

### CODE REVIEW

File: src/transformers/models/glpn/modeling_glpn.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def extra_repr(self) -> str:
 class GLPNOverlapPatchEmbeddings(nn.Module):
     """Construct the overlapping patch embeddings."""
 
-    def __init__(self, patch_size, stride, num_channels, hidden_size):
+    def __init__(self, config, patch_size, stride, num_channels, hidden_size):
```
Comment (ydshieh) [inline src/transformers/models/glpn/modeling_glpn.py:96]: Need this new argument so we can use `eps=config.layer_norm_eps`. As this is an internal class, should be fine.

File: src/transformers/models/segformer/modeling_segformer.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def extra_repr(self) -> str:
 class SegformerOverlapPatchEmbeddings(nn.Module):
     """Construct the overlapping patch embeddings."""
 
-    def __init__(self, patch_size, stride, num_channels, hidden_size):
+    def __init__(self, config, patch_size, stride, num_channels, hidden_size):
```
Comment (ydshieh) [inline src/transformers/models/segformer/modeling_segformer.py:127]: Need this new argument so we can use eps=config.layer_norm_eps. As this is an internal class, should be fine.

**Comment by NielsRogge** [comment]
Just to confirm:

- these changes don't have an impact on the integration tests, right? This only results in different outputs when one would train these models from scratch?
- technically we should check the layer_norm_eps for each paper to match the original code

**Comment by sgugger** [comment]
This is too breaking I think. We need to be more careful on new models added that this attribute is consistently used but I don't think we should touch old models like this as it will change the results of the forward.

**Comment by ydshieh** [comment]
OK! I will keep this list of models to skip in the WIP PR where we add a test for checking unused config attributes.

**Comment by ydshieh** [comment]
Close as it is too breaking!

**Pull request closed without merging**